ruma-events/tests/message_event.rs

```diff
@@ -222,3 +222,56 @@ fn deserialize_message_sticker() {
             && unsigned.is_empty()
     );
 }
+
+#[test]
+fn deserialize_message_then_convert_to_full() {
+    let rid = RoomId::try_from("!roomid:room.com").unwrap();
+    let json_data = json!({
+        "content": {
+            "answer": {
+                "type": "answer",
+                "sdp": "Hello"
+            },
+            "call_id": "foofoo",
+            "version": 1
+        },
+        "event_id": "$h29iv0s8:example.com",
+        "origin_server_ts": 1,
+        "sender": "@carl:example.com",
+        "type": "m.call.answer"
+    });
+
+    let sync_ev =
+        from_json_value::<Raw<AnySyncMessageEvent>>(json_data).unwrap().deserialize().unwrap();
+
+    // Test conversion method
+    let full = sync_ev.into_full_event(rid);
+    let full_json = to_json_value(full).unwrap();
+
+    assert_matches!(
+        from_json_value::<Raw<MessageEvent<AnyMessageEventContent>>>(full_json)
+            .unwrap()
+            .deserialize()
+            .unwrap(),
+        MessageEvent {
+            content: AnyMessageEventContent::CallAnswer(AnswerEventContent {
+                answer: SessionDescription {
+                    session_type: SessionDescriptionType::Answer,
+                    sdp,
+                },
+                call_id,
+                version,
+            }),
+            event_id,
+            origin_server_ts,
+            room_id,
+            sender,
+            unsigned,
+        } if sdp == "Hello" && call_id == "foofoo" && version == UInt::new(1).unwrap()
```

Comment:
Please put each of the checks on its own line. Also `UInt::new(1)` can just be `uint!(1)`.

Reply:
Do you want me to rewrite both files to use the string lits and `uint!`?

Reply:
All tests this PR adds or modifies should be updated accordingly.